File: backend/Minimarket.Api/Helpers/CryptoHelper.cs
using System.Security.Cryptography;
using System.Text;

namespace Minimarket.Api.Helpers;

// Cifrado simetrico (AES-256-CBC) para guardar secretos en la BD (ej. la clave de la IA).
// Compatible con el equivalente en Java (AES/CBC/PKCS5Padding, misma passphrase),
// para que ambos backends puedan descifrar el mismo valor.
public static class CryptoHelper
{
    private const string Passphrase = "Minimarket-Api-Secret-2026";

    private static byte[] Key => SHA256.HashData(Encoding.UTF8.GetBytes(Passphrase));

    public static string Encrypt(string plainText)
    {
        using var aes = Aes.Create();
        aes.Key = Key;
        aes.GenerateIV();

        using var encryptor = aes.CreateEncryptor();
        var plainBytes = Encoding.UTF8.GetBytes(plainText);
        var cipherBytes = encryptor.TransformFinalBlock(plainBytes, 0, plainBytes.Length);

        var result = new byte[aes.IV.Length + cipherBytes.Length];
        Buffer.BlockCopy(aes.IV, 0, result, 0, aes.IV.Length);
        Buffer.BlockCopy(cipherBytes, 0, result, aes.IV.Length, cipherBytes.Length);
        return Convert.ToBase64String(result);
    }

    public static string Decrypt(string cipherTextBase64)
    {
        var all = Convert.FromBase64String(cipherTextBase64);
        using var aes = Aes.Create();
        aes.Key = Key;

        var iv = new byte[16];
        Buffer.BlockCopy(all, 0, iv, 0, iv.Length);
        aes.IV = iv;

        using var decryptor = aes.CreateDecryptor();
        var plainBytes = decryptor.TransformFinalBlock(all, iv.Length, all.Length - iv.Length);
        return Encoding.UTF8.GetString(plainBytes);
    }
}

File: backend-java/minimarket-api-java/src/main/java/com/minimarket/api/util/CryptoHelper.java
package com.minimarket.api.util;

import javax.crypto.Cipher;
import javax.crypto.spec.IvParameterSpec;
import javax.crypto.spec.SecretKeySpec;
import java.nio.charset.StandardCharsets;
import java.security.MessageDigest;
import java.security.SecureRandom;
import java.util.Arrays;
import java.util.Base64;

// Cifrado simetrico (AES-256-CBC) para guardar secretos en la BD (ej. la clave de la IA).
// Compatible con el equivalente en .NET (misma passphrase y esquema IV+cipher en base64).
public final class CryptoHelper {

    private static final String PASSPHRASE = "Minimarket-Api-Secret-2026";

    private CryptoHelper() {
    }

    private static SecretKeySpec key() throws Exception {
        byte[] k = MessageDigest.getInstance("SHA-256").digest(PASSPHRASE.getBytes(StandardCharsets.UTF_8));
        return new SecretKeySpec(k, "AES");
    }

    public static String encrypt(String plainText) {
        try {
            byte[] iv = new byte[16];
            new SecureRandom().nextBytes(iv);
            Cipher cipher = Cipher.getInstance("AES/CBC/PKCS5Padding");
            cipher.init(Cipher.ENCRYPT_MODE, key(), new IvParameterSpec(iv));
            byte[] encrypted = cipher.doFinal(plainText.getBytes(StandardCharsets.UTF_8));
            byte[] out = new byte[iv.length + encrypted.length];
            System.arraycopy(iv, 0, out, 0, iv.length);
            System.arraycopy(encrypted, 0, out, iv.length, encrypted.length);
            return Base64.getEncoder().encodeToString(out);
        } catch (Exception ex) {
            throw new RuntimeException("No se pudo cifrar el valor.", ex);
        }
    }

    public static String decrypt(String cipherTextBase64) {
        try {
            byte[] all = Base64.getDecoder().decode(cipherTextBase64);
            byte[] iv = Arrays.copyOfRange(all, 0, 16);
            Cipher cipher = Cipher.getInstance("AES/CBC/PKCS5Padding");
            cipher.init(Cipher.DECRYPT_MODE, key(), new IvParameterSpec(iv));
            byte[] plain = cipher.doFinal(all, 16, all.length - 16);
            return new String(plain, StandardCharsets.UTF_8);
        } catch (Exception ex) {
            throw new RuntimeException("No se pudo descifrar el valor.", ex);
        }
    }
}
